app/controllers/provider/invitee_signups_controller.rb

```diff
@@ -49,7 +49,8 @@ def can_create?
   end
 
   def build_user
-    @user = @invitation.make_user(params[:user] || {})
+    permitted_params = params.permit(user: %i[password username])
```

Comment:
Are we sure the params are only those ones?

Reply:
I checked running specs and found these two params only

Reply:
The tests are probably lacking, we have more than those 2 fields, those are dynamic fields. You should look at the `show.html.erb` template

https://github.com/3scale/porta/blob/64b2cf8af119b89a480c7a7dd6d9fac1f06afc9d/app/views/provider/invitee_signups/show.html.erb#L22

And those are defined here https://github.com/3scale/porta/blob/64b2cf8af119b89a480c7a7dd6d9fac1f06afc9d/app/lib/fields/form_builder.rb#L13

which uses https://github.com/3scale/porta/blob/master/app/lib/fields/fields.rb#L241-L244

And you collect the `name` of those fields to permit here

Reply:
@hallelujah I checked the files and forms you mentioned, but not able to collect the filed names as they are not mentioned clearly.

can we permit all attributes instead? Or any easy and sure way I can collect name of fields names

Reply:
I described how to collect those names above basically `object.defined_fields.collect(&:name)`